apps/src/applab/designMode.js

```diff
@@ -587,6 +588,55 @@ designMode.onDuplicate = function(element, event) {
   return duplicateElement;
 };
 
+designMode.changeThemeForCurrentScreen = function(prevThemeValue, themeValue) {
+  const currentScreen = $(
+    elementUtils.getPrefixedElementById(
+      getStore().getState().screens.currentScreenId
+    )
+  );
+
+  // Unwrap the draggable wrappers around the elements in the source screen:
+  const madeUndraggable = makeUndraggable(currentScreen.children());
+
+  const screenAndChildren = [
+    currentScreen[0],
+    ...currentScreen.children().toArray()
+  ];
+
+  // Modify each element in the screen (including the screen itself):
+  screenAndChildren.forEach(element => {
+    const themeValues = elementLibrary.getThemeValues(element);
+    let modifiedProperty = false;
+    for (const propName in themeValues) {
+      const propTheme = themeValues[propName];
+      const prevDefault = propTheme[prevThemeValue];
+      const newDefault = propTheme[themeValue];
+      const currentPropValue = designMode.readProperty(element, propName);
+      const {type} = propTheme;
+      let propIsDefault = false;
+      if (type === 'color') {
+        propIsDefault =
+          new RGBColor(currentPropValue).toHex() ===
+          new RGBColor(prevDefault).toHex();
+      } else {
+        propIsDefault = currentPropValue === prevDefault;
+      }
+      if (propIsDefault) {
+        designMode.updateProperty(element, propName, newDefault);
+        modifiedProperty = true;
+      }
```

Comment:
Is there a helper we can extract for this inner loop?

Or maybe we want the function to be extracted on line 607:

```js
screenAndChildren.forEach(designMode.matchElementToCurrentTheme);
```

Reply:
Can we reuse or de-dup against `applyCurrentTheme` above?